### ISSUE TITLE
cmd: hwloc info showing extra cores

### ISSUE THREAD

**Issue by garrettbslone** [flux-framework/flux-core#2961]
I ran `flux mini run -n4 flux start flux hwloc info` after in a fresh flux instance of size one, and the command output was `4 Machines, 16 Cores, 32 PUs`.  Running `flux hwloc info` directly from the single node instance yields `1 Machine, 12 Cores, 24 PUs`. This was run on Ubuntu 20.04 with hwloc 2.0. Here is the output of `flux hwloc topology`: https://gist.github.com/garrettbslone/502696b8ce1052702e84cc4b6d32a9f7

**Comment by grondo** [comment]
Also interesting will be the `by_rank` summary for the single rank instance:

e.g.
```
$ flux kvs get resource.hwloc.by_rank
{"[0-3]": {"Package": 1, "Core": 4, "PU": 4, "cpuset": "0-3"}}
```

Then in the child instance, give us the `R` assigned to the job:
```
$ flux mini run -n4 flux start sh -c 'flux --parent job info ${FLUX_JOB_ID} R'
{"version":1,"execution":{"R_lite":[{"rank":"0","children":{"core":"0-3"}}]}}
```

And finally try a normal 4 task job and ensure cpu-affinity is working as expected

```
$ flux mini run -n4 sh -c 'taskset -c -p $$'
pid 30838's current affinity list: 0-3
pid 30836's current affinity list: 0-3
pid 30834's current affinity list: 0-3
pid 30832's current affinity list: 0-3
```

**Comment by garrettbslone** [comment]
Output of the command @grondo. It looks like the cpu-affinity may not me working properly.

`$ flux kvs get resource.hwloc.by_rank
{"0": {"Package": 1, "L3Cache": 4, "Core": 12, "PU": 24, "GPU": 1, "cpuset": "0-23"}}`

`$ flux mini run -n4 flux start sh -c 'flux --parent job info ${FLUX_JOB_ID} R'
{"version":1,"execution":{"R_lite":[{"rank":"0","children":{"core":"0-3"}}]}}`

`$ flux mini run -n4 sh -c 'taskset -c -p $$'
pid 390448's current affinity list: 0-3,12-15 
pid 390446's current affinity list: 0-3,12-15 
pid 390444's current affinity list: 0-3,12-15 
pid 390450's current affinity list: 0-3,12-15 
`

**Comment by grondo** [comment]
Ok, now try
```
$ flux mini run -n4 flux start flux kvs get resource.hwloc.by_rank
```

**Comment by garrettbslone** [comment]
```bash
$ flux mini run -n4 flux start flux kvs get resource.hwloc.by_rank
{"[0-3]": {"Package": 1, "L3Cache": 2, "Core": 4, "PU": 8, "GPU": 1, "cpuset": "0-3,12-15"}}
```

**Comment by grondo** [comment]
The above looks ok from a high level. 

  * You ask for 4 cores and you are getting a cpuset with 8 "cpus" -- presumably these are 8 PUs, 2 threads on each of 4 real cores

 * The "cpuset" being reported by `resource.hwloc.by_rank` is the union of all available cores as reported by libhwloc after `flux hwloc reload`. So presumably this would be consistent with hwloc.

 * The shell is using libhwloc to get the union of all core->cpuset for the assigned cores. I guess the mixup, if there is one, could be here if the shell is asking for logical cores and hwloc thinks we mean physical cores, or vice versa.

**Comment by grondo** [comment]
I wonder if this is related to #2965

**Comment by grondo** [comment]
@garrettbslone, not sure if this is 100% solved by #2966, but if so please feel free to close this issue.

**Comment by garrettbslone** [comment]
It seems as though it was a misunderstanding on my part.
